src/string_object.cpp

```diff
@@ -1959,4 +1959,29 @@ Value StringObject::chop_in_place(Env *env) {
     return this;
 }
 
+Value StringObject::sum(Env *env, Value val) {
+  int base = 16;
+  int sum = 0;
+  
+  if (val) {
+      if (!val->is_integer() && val->respond_to(env, "to_int"_s)){
+        val = val->send(env, "to_int"_s);
+      }
+      
+      if (val->is_integer()) {
```

Comment:
`assert_type()` validates that the type is an integer, which is what we want. You don't need the `if` around it.

Reply:
make sense 😅